model/metrics/dotnet-metrics.yaml

```
@@ -0,0 +1,213 @@
+groups:
+  - id: metric.dotnet.gc.collections
+    type: metric
+    metric_name: dotnet.gc.collections
+    brief: "Number of garbage collections that have occurred since the process has started."
+    note: "This metric reports the same values as calling [`GC.CollectionCount(int generation)`](https://learn.microsoft.com/dotnet/api/system.gc.collectioncount)."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.objects.size
+    type: metric
+    metric_name: dotnet.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    note: "This metric reports the same values as calling [`GC.GetTotalMemory(false)`](https://learn.microsoft.com/dotnet/api/system.gc.gettotalmemory)."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.gc.memory.total_allocated
+    type: metric
+    metric_name: dotnet.gc.memory.total_allocated
+    brief: >
+      The approximate number of bytes allocated on the managed GC heap since the process has
+      started. The returned value does not include any native allocations.
+    note: "This metric reports the same values as calling [`GC.GetTotalAllocatedBytes()`](https://learn.microsoft.com/dotnet/api/system.gc.gettotalallocatedbytes)."
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.gc.memory.commited
+    type: metric
+    metric_name: dotnet.gc.memory.commited
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
```

```suggestion
      The amount of committed virtual memory in use by the .NET GC, as
```
Maoni (.NET GC architect) suggested that the term 'heap' has a precise meaning in the context of the .NET GC. The GC heap includes all the memory used for storing managed objects and the fragmented space in between those objects. The GC heap does not include other committed memory which the GC holds as a cache to potentially grow the GC heap in the future. Since this committed size does include that cache memory its preferable not to call it the heap.

Also notice below the text says "Committed virtual memory may be larger than the heap size". That text is using .NET GC's precise meaning of heap which would contradict us refering to the whole thing as the heap here.